src/linalg/spectral.jl

```diff
@@ -151,13 +151,22 @@ function incidence_matrix(g::AbstractGraph, T::DataType=Int; oriented=false)
     # every col has the same 2 entries
     colpt = collect(1:2:(nz + 1))
     nzval = repeat([(isdir || oriented) ? -one(T) : one(T), one(T)], n_e)
+    swap(i,j) = begin
+        tmp = nzval[i]
+        nzval[i] = nzval[j]
+        nzval[j] = tmp
+    end
 
     # iterate over edges for row indices
     rowval = zeros(Int, nz)
     i = 1
-    for u in vertices(g)
-        for v in outneighbors(g, u)
+    for u in sort(vertices(g))
```

Comment:
`vertices(g)` is sorted by default for all SimpleGraphs. We make this assumption throughout the codebase so I'm not sure we need an explicit sort here.

Reply:
vertices is always a 1:n right? The compiler will optimize it out, but it will definitely break if we ever relax that assumption.

Reply:
Nice:

```
sort(r::AbstractUnitRange) = r
```